### PR TITLE
Preserve date and git commit prefix in tags.

### PR DESCRIPTION
**Pull request by aslonnie**
So the release tags are not changed when adding a cherrypick, like a doc cherrypick, and there will be no need to disable container builds manually.

We will retag the ones that we eventually want to release at the last moment.

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by aslonnie** [comment]
This will not affect Ray release tests today, as Ray release tests only use the wheel but not the container images (I think)

**Comment by aslonnie** [comment]
cc @jjyao

**Comment by aslonnie** [comment]
@ArturNiederfahrenhorst after this is merged, please cherrypick this into `releases/2.5.0` thanks; will make doc cherrypicks much easier.

**Comment by ArturNiederfahrenhorst** [comment]
@zhe-thoughts Please approve first.
@aslonnie If Zhe approves, please create the cherry pick yourself and enter it into go/r2.5picks!

**Comment by zhe-thoughts** [comment]
For merging to master, a Ray committer should review / approve / merge. Thanks

**Review by can-anyscale — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

**Review by krfricke — APPROVED** [review]

**Pull request merged**
